api/v1alpha1/addon_types.go

```diff
@@ -115,6 +115,11 @@ type CNI struct {
 	// +kubebuilder:default=HelmAddon
 	// +kubebuilder:validation:Enum=ClusterResourceSet;HelmAddon
 	Strategy *AddonStrategy `json:"strategy,omitempty"`
+
+	// ValuesRef is an object reference to Configmap/Secret which contains inline YAML
```

Comment:
```suggestion
	// ValuesRef is an object reference to Configmap which contains inline YAML
```
we could do the above to reduce the scope further.

Reply:
Perhaps not applicable for CNI configuration but secrets should be generally referenceable for values for any more confidential data.

Reply:
agree that we should use secrets for more confidential data in the values.
CAREN reads and parse the values from configmaps and set it as inline YAML in the HCP object. If we allow secret to be reference here, we have to decode user's secret and set it in the HCP inline YAML. Is it okay for CAREN to do this?

Reply:
any issue you foresee in CAREN doing it? One thing is CAREN needs to have permission to get secret but it would be pre-requisite so not much concerned. [Data format wise](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1011/files#diff-508d401bc31620f75e53a8c8a331155b1f9b13b815193f30de415fea48cdd97dR65-R98), there's no diff in CM v/s Secret

Reply:
The HCP excepts raw values in its CRD. We will be reading the secret data and setting it as plain text in HCP. I think we should not do that with secrets.

Reply:
Until HCP starts supporting secretRef, we'll need decoding in CAREN. Additional RBAC can be put on HCP, HRP, etc. to allow only legit users.

Later, even if HCP starts supporting secretRef, `helm get values <release-name>` on actual cluster will show all the helm values in plain text so there's no way to avoid this completely. The helm chart maintainer needs to design chart such that no sensitive data gets exposed maybe using external secret store, helm secret plugin, etc. but CAREN, CAAPH and helm has no part to play in this encryption/decryption.

Reply:
👍 .  RBAC should be good enough workaround.
My thoughts: The secrets should be generally referenced in the applications that are allowed to read in-memory. In our case, we ask users to provide secrets and write it plain text in the API server. This gives false sense of security. However RBAC should be fine until we have some native support for secrets in HCP / helm.

Reply:
> The helm chart maintainer needs to design chart such that no sensitive data gets exposed

Right; the Chart itself should support references to Secrets for any sensitive information. I _think_ the values should not contain sensitive information. If some Chart requires sensitive information in the values, then the Chart should be fixed. So referencing a Secret here may not be necessary.

As a counterpoint, though, Flux can [read values](https://github.com/fluxcd/pkg/blob/main/apis/meta/reference_types.go#L89-L93) from a Secret, perhaps to support Charts that expect sensitive information in their values.

@supershal w.r.t RBAC, do we want/need more restrictive rules for helm resources? :thinking:

Reply:
> @supershal w.r.t RBAC, do we want/need more restrictive rules for helm resources?

No. I think [current scope](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/blob/main/config/rbac/role_binding.yaml) should be good enough.